packages/react/src/profile/adapters/ProfileGateway.ts

```diff
@@ -44,4 +44,21 @@ export class ProfileGateway implements IProfileGateway {
       handle: data.result.handle,
     });
   }
+
+  async getProfileById(profileId: string): Promise<Profile | null> {
+    const { data } = await this.apolloClient.query<GetProfileQuery, GetProfileQueryVariables>({
+      query: GetProfileDocument,
+      variables: { request: { profileId } },
+    });
+
+    invariant(data, `Could not query profiles by id: ${profileId}`);
```

Comment:
nit: this is kinda tricky as we are shallowing the original error (the one in the `error` returned from the `apolloClient.query`). Can be confusing later to debug real production errors as only `Could not query profiles by id` message is there without a real reason.
Not saying we have to solve it now, just building awareness.

Reply:
I agree, and this is not the only place where this happens. I created a ticket to discuss this issue: T-4022